### PR TITLE
chore: pre-commit formatting

### PR DESCRIPTION
**Pull request by Anush008**
Updated the pre-commit config to include Ruff and ran it on all the files.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+    rev: 23.1.0
+    hooks:
+      - id: black
+        name: "Black: The uncompromising Python code formatter"
+
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        name: "Sort Imports"
```
Comment (NirantK) [inline .pre-commit-config.yaml:19]: Wouldn't want to have these. Isort is already covered in ruff. And so is most of what we need from black.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.13
```
Comment (NirantK) [inline .pre-commit-config.yaml:21]: This version is different from our pyproject.toml — please sync them?

**Review by NirantK — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (Anush008) [reply .pre-commit-config.yaml:3]: https://github.com/qdrant/fastembed/commit/47ee7ad84cbd7e96774b22180d889c7fe1e434b2

Comment (Anush008) [reply .pre-commit-config.yaml:3]: https://github.com/qdrant/fastembed/commit/5b66b70eb12c2dc14e295c3d92920592d5daf327

Comment (NirantK) [reply .pre-commit-config.yaml:3]: This is still out of sync: https://github.com/qdrant/fastembed/blob/b01f882df7113b24b37391a8ada65ba5e4886bb7/pyproject.toml#L24

**Review by NirantK — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -21,9 +21,7 @@ huggingface-hub = "0.19.4"
 
 [tool.poetry.group.dev.dependencies]
 pytest = "^7.4.2"
-ruff = "^0.0.277"
-isort = "^5.12.0"
-black = "^23.7.0"
+ruff = "^0.1.13"
```
Comment (Anush008) [inline pyproject.toml:24]: @NirantK

**Review by NirantK — APPROVED** [review]
Thanks for doing this! 

Squash and Merge when tests clear

**Pull request merged**